src/netex-convertor/netexGenerator.ts

```diff
@@ -94,10 +106,11 @@ const netexGenerator = (
         } else {
             // update period only
             publicationRequestToUpdate.topics.NetworkFrameTopic.TypeOfFrameRef.ref = `fxc:UK:DFT:TypeOfFrame_UK_PI_${
-                isGeoZoneTicket(ticket) ? 'NETWORK' : 'LINE'
+                isGeoZoneTicket(ticket) || isSchemeOperatorGeoZoneTicket(ticket) ? 'NETWORK' : 'LINE'
             }_FARE_OFFER:FXCP`;
 
-            publicationRequestToUpdate.topics.NetworkFrameTopic.NetworkFilterByValue.objectReferences.PreassignedFareProductRef = ticket.products.map(
+            const products = ticket.products as Array<ProductDetails | FlatFareProductDetails>;
```

Comment:
I think we can get rid of this assertion, when mapping over a union type you need to explicitly set the types though so line 114 in the map would need to be `(product: ProductDetails | FlatFareProductDetails) => ({...`